### ISSUE TITLE
Slug does not have a linked field

### ISSUE THREAD

**Issue by maartenvn** [directus/next#437]
**Directus**: 9.0.0-beta.2
**Node**: v14.10.0
**MySQL**: Ver 14.14

There is no option to link a slug to an existing field.

![image](https://user-images.githubusercontent.com/6013068/94472503-24a95c80-01cb-11eb-819c-23b8b3bc3bff.png)

**Comment by maartenvn** [comment]
https://github.com/directus/next/blob/282746a68fc7215bcdb3b8582860a5441c729d58/app/src/interfaces/slug/index.ts#L11

There is no option for "Mirrored Field" like in v8.
Maybe instead of a string like in v8 a dropdown could be used to select the field.
Or even better would be to use `{{ mustache }}`-syntax so you could have the option to use multiple fields:

- eg: `{{ title }}-{{ id }}`

**Comment by rijkvanzanten** [comment]
We're trying to think through how to implement this in a way that the API can enforce it as well. By adding an interface option and handling it on the client side, this would only work when using the Directus app to create an item, but will be ignored when creating an item through the API directly

**Comment by nickrum** [comment]
@rijkvanzanten have you considered implementing this combined with #353?

**Comment by rijkvanzanten** [comment]
@nickrum Yes and no. Calculated fields could be used to dynamically generate out a slug whenever you _read_ an item, but it won't help generating and saving a slug on _create_ 🤔

**Comment by nickrum** [comment]
But does it really have to be generated and saved on _create_? Wouldn't it be enough if a slug is just a dynamically generated alias? The only downside i can see is that slugs couldn't be set manually anymore which could be mitigated by additionally using a regular string field (and possibly some Interface to help with the UX). Maybe I am overlooking the obvious here 🤔

**Comment by rijkvanzanten** [comment]
@nickrum The problem with dynamically generated slugs is that they change whenever the linked field changes. With a slug-on-create, you only generate the slug once, which ensures that you don't end up with dead links in your project whenever the actual linked field changes

**Comment by nickrum** [comment]
Oh I see, that's unfortunate. Thanks for the explanation @rijkvanzanten.
